shipyard-controller/internal/handler/debugmanager.go

```diff
@@ -90,3 +91,66 @@ func (dm *DebugManager) GetEventByID(projectName string, shkeptncontext string,
 func (dm *DebugManager) GetAllProjects() ([]*apimodels.ExpandedProject, error) {
 	return dm.projectRepo.GetProjects()
 }
+
+func (dm *DebugManager) GetBlockingSequences(projectName string, shkeptncontext string, stage string) ([]models.SequenceExecution, error) {
+
+	_, err := dm.projectRepo.GetProject(projectName)
+	if err != nil {
+		return nil, err
+	}
+
+	sequences, err := dm.sequenceExecutionRepo.Get(models.SequenceExecutionFilter{
+		Scope: models.EventScope{
+			KeptnContext: shkeptncontext,
+			EventData: keptnv2.EventData{
+				Project: projectName,
+				Stage:   stage,
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sequences) == 0 {
+		return nil, common.ErrSequenceNotFound
+	}
+
+	sequence := sequences[0]
```

Comment:
There could be multiple sequences on the same stage for the same project with the same `shkeptncontext`.
You should sort it by time similar to the following snippet and take the latest one.
```
sort.Slice(sequences, func(i, j int) bool {
	return events[i].Time.After(events[j].Time)
})
```